compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -794,6 +794,8 @@ public void visit(BLangPackage pkgNode) {
 
         pkgNode.constants = removeDuplicateConstants(pkgNode);
 
+        rewrite(pkgNode.typeDefinitions, env);
```

Comment:
Don't you need the sorting by precedence done  in L809 before this?

Reply:
I checked that. What I understand is After desugaring also, we can sort that (that sort is needed only for annotations as I understand).